packages/core/contracts/financial-templates/optimistic-rewarder/OptimisticRewarderCreator.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../common/implementation/Lockable.sol";
+import "../../common/implementation/Testable.sol";
+
+import "../../oracle/interfaces/FinderInterface.sol";
+import "./OptimisticRewarder.sol";
+
+/**
+ * @notice The creator contract for optimistic rewarders. Using this contract is totally optional. It only aids in
+ * creating a simpler deployment experience with a guarantee of repeatable verification and easier tracking through
+ * events.
+ */
+contract OptimisticRewarderCreator is Lockable {
+    FinderInterface public finder;
+
+    event CreatedOptimisticRewarder(address indexed optimisticRewarder, bool includesToken);
+
+    constructor(FinderInterface _finder) {
+        finder = _finder;
+    }
+
+    /**
+     * @notice Deploys an optimistic rewarder.
+     * @param _name name for the ERC721 token.
+     * @param _symbol symbol for the ERC721 token.
+     * @param _baseUri prefix to each ERC721 tokenId's name.
+     * @param _liveness liveness period between submission and verification of a reward.
+     * @param _bondToken ERC20 token that the bond is paid in.
+     * @param _bond size of the bond.
+     * @param _identifier identifier that should be passed to the optimistic oracle on dispute.
+     * @param _customAncillaryData custom ancillary data that should be sent to the optimistic oracle on dispute.
+     */
+    function createOptimisticRewarder(
```

Comment:
very nice that we are not needing to use a struct any more for these params.

Reply:
Agreed!